lib/charms/opensearch/v0/opensearch_plugins.py

```diff
@@ -350,7 +350,78 @@ def disable(self) -> OpenSearchPluginConfig:
             config_entries_to_add={"knn.plugin.enabled": False},
         )
 
+    @property
+    def dependencies(self) -> Optional[List[str]]:
+        """Returns a list of plugin name dependencies."""
+        return ["opensearch-ml"]
+
     @property
     def name(self) -> str:
         """Returns the name of the plugin."""
         return "opensearch-knn"
+
+
+class OpenSearchPluginMLCommons(OpenSearchPlugin):
+    """Implements the opensearch-ml-commons plugin."""
+
+    def config(self) -> OpenSearchPluginConfig:
+        """Returns a plugin config object to be applied for enabling the current plugin."""
+        return OpenSearchPluginConfig(
+            config_entries_to_add={
+                "plugins.ml_commons.only_run_on_ml_node": True,
+                "plugins.ml_commons.task_dispatch_policy": "round_robin",
+                "plugins.ml_commons.max_ml_task_per_node": 20,
+                "plugins.ml_commons.max_model_on_node": 100,
+                "plugins.ml_commons.sync_up_job_interval_in_seconds": 3,
+                "plugins.ml_commons.monitoring_request_count": 100,
+                "plugins.ml_commons.max_register_model_tasks_per_node": 10,
+                "plugins.ml_commons.allow_registering_model_via_url": False,
+                "plugins.ml_commons.allow_registering_model_via_local_file": False,
+                "plugins.ml_commons.trusted_url_regex": '"^(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]"',
+                "plugins.ml_commons.ml_task_timeout_in_seconds": 600,
+                "plugins.ml_commons.native_memory_threshold": 90,
+                "plugins.ml_commons.allow_custom_deployment_plan": False,
+                "plugins.ml_commons.model_auto_redeploy.lifetime_retry_times": 3,
+                "plugins.ml_commons.model_auto_redeploy_success_ratio": 0.8,
+                "plugins.ml_commons.enable_inhouse_python_model": False,
+                # Enablement settings
+                "plugins.ml_commons.remote_inference.enabled": True,
+                "plugins.ml_commons.model_access_control_enabled": False,
+                "plugins.ml_commons.connector_access_control_enabled": False,
+                "plugins.ml_commons.model_auto_redeploy.enable": False,
+            },
+        )
+
+    def disable(self) -> OpenSearchPluginConfig:
+        """Returns a plugin config object to be applied for disabling the current plugin."""
+        return OpenSearchPluginConfig(
+            config_entries_to_add={
+                "plugins.ml_commons.remote_inference.enabled": False,
+                "plugins.ml_commons.model_access_control_enabled": False,
+                "plugins.ml_commons.connector_access_control_enabled": False,
+                "plugins.ml_commons.model_auto_redeploy.enable": False,
```

Comment:
we can delete these, if they were initially set to `True` we could have just list them as part of `config_entries_to_del` to set them back to their default value (False)